platforms/core-runtime/base-services/build.gradle.kts

```diff
@@ -13,6 +13,12 @@ gradlebuildJava.usedInWorkers()
 tasks.named<JavaCompile>("compileTestJava") {
     options.release = 8
 }
+afterEvaluate {
```

Comment:
afterEvaluate is annoying here, since the unittest-and-compile plugin uses afterEvaluate. 

In practice, the plugin should be enhanced to allow projects to specify the java versions for their tests.

In any case, this will go away in 9.0, as the project will target Java 8 anyway.